docker-compose-developer.yml

```diff
@@ -1,80 +1,232 @@
+    command: --default-authentication-plugin=mysql_native_password --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci
+    restart: always
+    environment:
+      MYSQL_ROOT_PASSWORD: empty
+      MYSQL_DATABASE: oncall_local_dev
     ports:
-      - "15672:15672"
-      - "5672:5672"
+      - "3306:3306"
+    deploy:
+      resources:
+        limits:
+          memory: 500m
+          cpus: "0.5"
+    healthcheck:
+      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
+      timeout: 20s
+      retries: 10
+    volumes:
+      - mysqldata:/var/lib/mysql
+    profiles:
+      - mysql
 
-  mysql-to-create-grafana-db:
+  mysql_to_create_grafana_db:
+    container_name: mysql_to_create_grafana_db
     image: mysql:5.7
     platform: linux/x86_64
     command: bash -c "mysql -h mysql -uroot -pempty -e 'CREATE DATABASE IF NOT EXISTS grafana CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;'"
     depends_on:
       mysql:
         condition: service_healthy
+    profiles:
+      - mysql
+
+  postgres:
+    container_name: postgres
+    image: postgres:14.4
+    restart: always
+    environment:
+      POSTGRES_DB: oncall_local_dev
+      POSTGRES_PASSWORD: empty
+      POSTGRES_INITDB_ARGS: --encoding=UTF-8
+    ports:
+      - "5432:5432"
+    deploy:
+      resources:
+        limits:
+          memory: 500m
+          cpus: "0.5"
+    healthcheck:
+      test: ["CMD", "pg_isready", "-U", "postgres"]
+      interval: 10s
+      timeout: 5s
+      retries: 5
+    volumes:
+      - postgresdata:/var/lib/postgresql/data
+    profiles:
+      - postgres
+
+  postgres_to_create_grafana_db:
+    container_name: postgres_to_create_grafana_db
+    image: postgres:14.4
+    command: bash -c "PGPASSWORD=empty psql -U postgres -h postgres -tc \"SELECT 1 FROM pg_database WHERE datname = 'grafana'\" | grep -q 1 || PGPASSWORD=empty psql -U postgres -h postgres -c \"CREATE DATABASE grafana\""
+    depends_on:
+      postgres:
+        condition: service_healthy
+    profiles:
+      - postgres
 
   grafana:
+    container_name: grafana
     image: "grafana/grafana:main"
     restart: always
     environment:
-      GF_DATABASE_TYPE: mysql
-      GF_DATABASE_HOST: mysql
-      GF_DATABASE_USER: root
-      GF_DATABASE_PASSWORD: empty
       GF_SECURITY_ADMIN_USER: oncall
       GF_SECURITY_ADMIN_PASSWORD: oncall
       GF_PLUGINS_ALLOW_LOADING_UNSIGNED_PLUGINS: grafana-oncall-app
+    env_file:
+      - .env.${DB}.dev
+    ports:
+      - "3000:3000"
     deploy:
       resources:
         limits:
           memory: 500m
-          cpus: '0.5'
+          cpus: "0.5"
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
     volumes:
+      - grafana_data:/var/lib/grafana
       - ./grafana-plugin:/var/lib/grafana/plugins/grafana-plugin
-    ports:
-      - "3000:3000"
-    depends_on:
-      mysql:
-        condition: service_healthy
+    profiles:
+      - grafana
+
+volumes:
+  redisdata:
+  grafana_data:
```

Comment:
nit: all other volume names don't have an underscore, consider making this consistent.
Also we could probably add a `_dev` postfix for these volumes so they're not mixed up with the volumes used for hobby OSS environment.